client/state/data-layer/wpcom/read/tags/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_TAGS_REQUEST,
+} from 'state/action-types';
+import {
+	receiveTags,
+} from 'state/reader/tags/items/actions';
+
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export function requestTags( store, action, next ) {
+	const path = action.payload && action.payload.slug
+		? `/read/tags/${ action.payload.slug }`
+		: '/read/tags';
+
+	store.dispatch( http( {
+		path,
+		method: 'GET',
+		apiVersion: '1.2',
+		onSuccess: action,
+		onFailure: action,
+	} ) );
+
+	next( action );
+}
+
+/**
+ * Normalize response from the api so whether we get back a single tag or a list of tags
+ * we always pass forward a list
+ * @param  {Tag|Tags} apiResponse api response from the tags endpoint
+ * @return {Tags}             An object containing list of tags
```

Comment:
Are these data types defined somewhere? Also, what is `Tags`? Is it supposed to be a list of tags? Check out the [@type](http://usejsdoc.org/tags-type.html) property for JSDoc

```js
/**
 * @typedef {Object} Tag
 * @property {String} name Name of tag?
 */

/**
 * @typedef {Object} TagApiResponse
 * @property {?Tag} tag if a single tag exists?
 * @property {?Tag[]} tags if multiple tags exist?
 */

@param {TagApiResponse}
```